### PR TITLE
Limit max tip age to reasonable value

### PR DESCRIPTION
**Pull request by bvbfan**
/kind fix

#### What this PR does / why we need it:

Max tip age is used in IBD it allows miners to start their work too early which can result in high hash error.
It's a safe value since leaving IBD is happen one time per run we cannot end with stalling masternode.
Fee estimation tip age is in direct connection to max tip age and should be lowered as well.

### CODE REVIEW

File: src/validation.h [hunk 1]
```diff
@@ -609,6 +601,14 @@ class CChainState {
...
     CChainState();
 
+    /**
+    * Whether this chainstate is undergoing initial block download.
+    *
+    * Mutable because we need to be able to mark IsInitialBlockDownload()
+    * const, which latches this for caching purposes.
+    */
+    mutable std::atomic<bool> m_cached_finished_ibd{false};
+
```
Comment (ShengguangXiao) [inline src/validation.h:611]: Why don't keep the variable private and add a function to set it?

Comment (Bushstar) [reply src/validation.h:611]: Probably sensible but this was done in a rush yesterday and this PR is to provide a temp solution to keep help stakers.

File: src/init.cpp [hunk 2]
```diff
@@ -1203,6 +1203,7 @@ bool AppInitParameterInteraction()
         return InitError("unknown rpcserialversion requested.");
 
     nMaxTipAge = gArgs.GetArg("-maxtipage", DEFAULT_MAX_TIP_AGE);
+    nMaxTipAge = std::min(nMaxTipAge, DEFAULT_MAX_TIP_AGE);
```
Comment (ShengguangXiao) [inline src/init.cpp:1206]: Do we still limit the value to 10 minutes?

Comment (bvbfan) [reply src/init.cpp:1206]: We can remove it.

**Review by Bushstar — APPROVED** [review]

**Pull request merged**